cirq-google/cirq_google/ops/fsim_gate_family.py

```diff
@@ -0,0 +1,385 @@
+        atol=DEFAULT_ATOL,
+    ):
+        """Inits `cirq_google.FSimGateFamily`.
+
+        Args:
+            gates_to_accept: List of gate types or instances to be accepted. All elements
+                should be either types from `POSSIBLE_FSIM_GATES` or non-parameterized
+                instances of those types.
+            gate_types_to_check: List of `POSSIBLE_FSIM_GATES` types whose instances should be
+                considered when trying to match against gates present in `gates_to_accept`.
+                Defaults to all `POSSIBLE_FSIM_GATES` if left unspecified.
+            allow_symbols: If True, both the gate conversion logic and containment predicate allow
+                parameterized gate instances and return a converted gate / accept input gate if
+                there exists any value of the unknown parameters which can result in a valid
+                outcome.
+            atol: Absolute tolerance for difference floating point comparisons.
+
+        Raises:
+            ValueError: If any gate in `gates_to_accept` is not a non-parameterized instance of /
+                or a gate type from `POSSIBLE_FSIM_GATES`.
+            ValueError: If any gate type in `gate_types_to_check` is not one of
+                `POSSIBLE_FSIM_GATES`.
+        """
+        self._supported_types: Dict[
+            Type[POSSIBLE_FSIM_GATES],
+            Callable[[POSSIBLE_FSIM_GATES], Optional[POSSIBLE_FSIM_GATES]],
+        ] = {
+            cirq.FSimGate: self._convert_to_fsim,
+            cirq.PhasedFSimGate: self._convert_to_phased_fsim,
+            cirq.ISwapPowGate: self._convert_to_iswap,
+            cirq.PhasedISwapPowGate: self._convert_to_phased_iswap,
+            cirq.CZPowGate: self._convert_to_cz,
+            cirq.IdentityGate: self._convert_to_identity,
+        }
+        if not gate_types_to_check:
+            gate_types_to_check = tuple(self._supported_types.keys())
+
+        if any(g not in self._supported_types for g in gate_types_to_check):
+            raise ValueError(
+                f"All gates in gate_types_to_check: {_gate_list_to_str(gate_types_to_check)} must "
+                f"be one of {_gate_list_to_str(self._supported_types.keys())}."
+            )
+
+        for g in gates_to_accept:
+            if isinstance(g, tuple(self._supported_types.keys())):
+                if cirq.is_parameterized(g):
+                    raise ValueError(
+                        f"Parameterized gate {g} cannot be used in `gates_to_accept` initializer."
+                    )
+            elif g not in self._supported_types:
+                raise ValueError(
+                    f"Gate {g} in `gates_to_accept` must be either a type from or an instance of "
+                    f"{_gate_list_to_str(self._supported_types.keys())}"
+                )
+        self.gates_to_accept = tuple(dict.fromkeys(gates_to_accept))
+        self.gate_types_to_check = tuple(dict.fromkeys(gate_types_to_check))
```

Comment:
Seems odd here to convert to dict but then back to tuple which iterates over the dict keys and ignores the value anyway. I'd suggest converting to `set` instead:
```suggestion
        self.gates_to_accept = tuple(set(gates_to_accept))
        self.gate_types_to_check = tuple(set(gate_types_to_check))
```
Or just use `frozenset` directly instead of a tuple:
```suggestion
        self.gates_to_accept = frozenset(gates_to_accept)
        self.gate_types_to_check = frozenset(gate_types_to_check)
```

Reply:
The goal here is to keep only the unique values from the input but also preserve the input ordering so that repr/str look as expected to the users.  Iterable --> dict --> tuple is a succinct way to achieve this.  

This has also been done earlier for gatesets, which had a similar requirement: 
https://github.com/quantumlib/Cirq/blob/54286063f679d67501ff1b905cd16b879feaae27/cirq-core/cirq/ops/gateset.py#L215